vms/platformvm/docs/validators_versioning.md

```diff
@@ -0,0 +1,113 @@
+
+## Validator diffs content
+
+Every new block accepted by the P-chain can potentially alter the validator set of any Subnet, including the primary one. New validators may be added; some of them may have reached their end of life and are therefore removed. Moreover a validator can register itself again once its staking time is done, possibly with a `Weight` and a `BLS Public key` different from the previous staking period.
+
+Whenever the block at height `H` adds or removes a validator, the P-chain does, among others, the following operations:
+
+1. it updates the current validator set to add the new validator or remove it if expired;
+2. it explicitly records the validator set diffs with respect to the validator set at height `H-1`.
+
+These diffs are key to rebuilding the validator set at a given past height. In this section we illustrate their content. In next ones, We'll see how the diffs are stored and used.
+
+The validators diffs track changes in a validator's `Weight` and `BLS Public key`. Along with the `NodeID` this is the data exposed by the `GetValidatorSet` method.
+
+Note that `Weight` and `BLS Public key` behave differently throughout the validator lifetime:
+
+1. `BLS Public key` cannot change through a validator's lifetime. It can only change when a validator is added/re-added and removed.
+2. `Weight` can change throughout a validator's lifetime by the creation and removal of its delegators as well as by validator's own creation and removal.
+
+Here is a scheme of what `Weight` and `BLS Public key` diff content we record upon relevant scenarios: 
+
+|                    | Weight Diff (forward looking)                                                                           | BLS Key Diff (backward looking)                                                       |
+|--------------------|---------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------|
+| Validator creation | record ``` golang state.ValidatorWeightDiff{       Decrease: false,     Weight: validator.Weight, } ``` | record an empty byte slice if validator.BlsKey is specified; otherwise record nothing |
+| Delegator creation | record ``` golang state.ValidatorWeightDiff{      Decrease: false,     Weight: validator.Weight, } ```  | No entry is recorded                                                                  |
+| Delegator removal  | record ``` golang state.ValidatorWeightDiff{      Decrease: true,     Weight: validator.Weight, } ```  | No entry is recorded                                                                  |
+| Validator removal  | record ``` golang state.ValidatorWeightDiff{      Decrease: true,     Weight: validator.Weight, } ```  | record validator.BlsKey if it is specified; otherwise record nothing                  |
```

Comment:
```suggestion
| Validator creation | record ```golang state.ValidatorWeightDiff{      Decrease: false,     Weight: validator.Weight, } ```  | record an empty byte slice if validator.BlsKey is specified; otherwise record nothing |
| Delegator creation | record ```golang state.ValidatorWeightDiff{      Decrease: false,     Weight: validator.Weight, } ```  | No entry is recorded                                                                  |
| Delegator removal  | record ```golang state.ValidatorWeightDiff{      Decrease: true,     Weight: validator.Weight, } ```  | No entry is recorded                                                                  |
| Validator removal  | record ```golang state.ValidatorWeightDiff{      Decrease: true,     Weight: validator.Weight, } ```  | record validator.BlsKey if it is specified; otherwise record nothing                  |
```